packages/orbit-components/src/Itinerary/context.jsx

```diff
@@ -0,0 +1,33 @@
+// @flow
+import * as React from "react";
+
+import type { Context } from "./context";
+import typeof { ItineraryProvider as ItineraryProviderType } from "./context";
+
+export const ItineraryContext: React.Context<Context> = React.createContext({
+  setWidths: () => {},
+  setCalculatedWidth: () => {},
+  calculatedWidth: 0,
+});
+
+export const ItineraryProvider: ItineraryProviderType = ({ children }): React.Node => {
+  const [widths, setWidths] = React.useState([]);
+  const [calculatedWidth, setCalculatedWidth] = React.useState(0);
+
+  React.useLayoutEffect(() => {
```

Comment:
Unless you're reading layout from the DOM you shouldn't need `useLayoutEffect`.
```suggestion
  React.useEffect(() => {
```